agent/hcp/telemetry/filter.go

```diff
@@ -0,0 +1,51 @@
+package telemetry
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// filterList holds a map of filters, i.e. regular expressions.
+// These filters are used to identify which Consul metrics can be transmitted to HCP.
+type filterList struct {
+	isValid *regexp.Regexp
+}
+
+// newFilterList returns a FilterList which holds valid regex used to filter metrics.
+// It will fail if there are 0 valid regex filters given.
+func newFilterList(filters []string) (*filterList, error) {
```

Comment:
Why not just return `*regexp.Regexp`?

Reply:
Done, thanks! It was a different implementation earlier, which was simplified so it makes more sense to just return a regex